internal/controller/etcdopstask/utils.go

```diff
@@ -0,0 +1,308 @@
+}
+
+// taskStatusUpdate represents a status update for a task
+type taskStatusUpdate struct {
+	// Operation details to update (nil to skip operation update)
+	Operation *druidv1alpha1.LastOperation
+	// Task state to set (nil to skip state update)
+	State *druidv1alpha1.TaskState
+	// Phase to set (nil to skip phase update)
+	Phase *druidv1alpha1.OperationPhase
+	// Error to record (nil to skip error recording)
+	Error error
+}
+
+// setLastOperation updates the LastOperation field in the task status
+func setLastOperation(task *druidv1alpha1.EtcdOpsTask, opType druidv1alpha1.LastOperationType, state druidv1alpha1.LastOperationState, description string, runID string) {
+	now := metav1.Time{Time: time.Now().UTC()}
+	if description == "" {
+		description = fmt.Sprintf("%s is in state %s for task %s", opType, state, task.Name)
+	}
+
+	if task.Status.LastOperation == nil {
+		task.Status.LastOperation = &druidv1alpha1.LastOperation{
+			Type:           opType,
+			State:          state,
+			LastUpdateTime: now,
+			Description:    description,
+			RunID:          runID,
+		}
+		return
+	}
+
+	task.Status.LastOperation.Type = opType
+	task.Status.LastOperation.State = state
+	task.Status.LastOperation.LastUpdateTime = now
+	task.Status.LastOperation.Description = description
+	task.Status.LastOperation.RunID = runID
+}
+
+// setTaskState updates the task's State field
+func setTaskState(task *druidv1alpha1.EtcdOpsTask, state druidv1alpha1.TaskState) bool {
+	stateChanged := task.Status.State == nil || *task.Status.State != state
+	if !stateChanged {
+		return false
+	}
+
+	now := &metav1.Time{Time: time.Now().UTC()}
+	if state == druidv1alpha1.TaskStateInProgress && task.Status.StartedAt == nil {
+		task.Status.StartedAt = now
+	}
+	task.Status.State = &state
+	task.Status.LastTransitionTime = now
+	return true
+}
+
+// setPhase updates the task's Phase field
+func setPhase(task *druidv1alpha1.EtcdOpsTask, phase druidv1alpha1.OperationPhase) bool {
+	phaseChanged := task.Status.Phase == nil || *task.Status.Phase != phase
+	if !phaseChanged {
+		return false
+	}
+
+	task.Status.Phase = &phase
+	return true
+}
+
+// setLastError adds an error to the LastErrors field
+func setLastError(task *druidv1alpha1.EtcdOpsTask, err error) {
+	now := metav1.Time{Time: time.Now().UTC()}
+
+	newErrors := druiderr.MapToLastErrors([]error{err})
+	if len(newErrors) == 0 {
+		newErrors = []druidv1alpha1.LastError{{
+			Description: err.Error(),
+			ObservedAt:  now,
+		}}
+	} else {
+		newErrors[0].ObservedAt = now
+	}
+
+	lastErrors := task.Status.LastErrors
+	if lastErrors == nil {
+		lastErrors = make([]druidv1alpha1.LastError, 0, 3)
+	}
+	if len(lastErrors) >= 3 {
+		lastErrors = lastErrors[1:]
+	}
+
+	task.Status.LastErrors = append(lastErrors, newErrors[0])
+}
+
+// updateTaskStatus updates operation, state, and errors in a single call.
+func (r *Reconciler) updateTaskStatus(ctx context.Context, task *druidv1alpha1.EtcdOpsTask, update taskStatusUpdate) error {
```

Comment:
is it not possible to create a clone of EtcdOpsTask and update its status and then this function is no longer required, right? What is the need for this function?